pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go

```diff
@@ -250,6 +260,9 @@ func (cra *cpuResourceAdvisor) update() {
 		r.TryUpdateHeadroom()
 		cra.updateRegionHeadroom()
 	}
+	if enableIsolated && isolationExists && !cra.checkIsolationSafety() {
+		return false
```

Comment:
I'm not sure if there is any race between twice updateWithIsolationGuardians.

Reply:
we have a locking  logic`cra.mutex.Lock()`